src/pages/SmartWeaveExplorer/Contract/Contract.vue

```diff
@@ -580,22 +596,26 @@ export default {
         this.getInteractions(this.currentPage, this.selected);
       }
     },
+
     async getContract() {
-      axios
-        .get(`${this.gatewayUrl}/gateway/contract?txId=${this.contractId}`)
-        .then((fetchedContract) => {
-          this.owner = fetchedContract.data.owner;
-          this.pst_ticker = fetchedContract.data.pstTicker;
-          this.pst_name = fetchedContract.data.pstName;
-          this.wasmLang = fetchedContract.data.srcWasmLang;
-          this.initState = fetchedContract.data.initState;
-          this.loadedContract = true;
-          this.sourceTxId = fetchedContract.data.srcTxId;
-        })
-        .catch((e) => {
-          this.correct = false;
-        });
+      const response = await fetch(`${this.gatewayUrl}/gateway/contract?txId=${this.contractId}`);
+      if (!response.ok) {
+        this.correct = false;
+        const error = `An error has occured: ${response.status}`;
+        throw new Error(error);
+      }
+      const data = await response.json();
+
+      this.tags = await interactionTagsParser(data);
```

Comment:
check out this contract `NwaSMGCdz6Yu5vNjlMtCNBmfEkjYfT-dfYkbQQDGn5s` - somehow contractTx is returned as null from the gateway. You should consider such case and display a warning message , sth like 'Contract tags are not available.'. I'll talk with ppe to discuss why contractTx is null as its kind of weird.

Reply:
I've added an error 'Tags are unavailable' in tags tab, if tag array is empty